src/Analyzers/CSharp/CodeFixes/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -87,24 +87,27 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             var currentRoot = root.TrackNodes(nodesToTrack);
 
             var languageVersion = semanticModel.SyntaxTree.Options.LanguageVersion();
-            bool makeStaticIfPossible;
+            var makeStaticIfPossible = false;
 
             if (languageVersion >= LanguageVersion.CSharp8)
             {
-                var options = (CSharpCodeGenerationOptions)await document.GetCodeGenerationOptionsAsync(fallbackOptions, cancellationToken).ConfigureAwait(false);
+#if CODE_STYLE
+                var info = new CSharpCodeGenerationContextInfo(
+                    CodeGenerationContext.Default, CSharpCodeGenerationOptions.Default, new CSharpCodeGenerationService(document.Project.Services), root.SyntaxTree.Options.LanguageVersion());
+#else
+                var info = await document.GetCodeGenerationInfoAsync(CodeGenerationContext.Default, fallbackOptions, cancellationToken).ConfigureAwait(false);
+#endif
+
+                var options = (CSharpCodeGenerationOptions)info.Options;
```

Comment:
@tmat is this the right way to handle this pattern?  or is there one simple mechanism to get this information for both layers?